inbox_ui: Clarify inbox state when all sections are collapsed.

whilstsomebody opened the pull request:
It was reported that the UI is pretty confusing when there is just a collapsed folder in inbox view. To fix this, this PR does the following two tasks: 

- [x] Adds a note below the folders which says "All of your unread conversations are hidden. Click on a section, folder, or channel to see what's inside.
- [x] The note is displayed for the following situtation:
       -  All folders collapsed.
       -  If all folders are not collapsed, all channels are collapsed (all visible channels should be collapsed).
- [x] Adds an "Show all" button to expand all the collapsed folders at once.

<img width="950" height="310" alt="image" src="https://github.com/user-attachments/assets/9a37a41d-375d-44f2-80e8-c24e5258b112" />  
<img width="911" height="286" alt="image" src="https://github.com/user-attachments/assets/b3e4fa2d-a6fa-48db-aee1-0337f552332e" />

</br>

<img width="950" height="481" alt="image" src="https://github.com/user-attachments/assets/d5c663f8-20b7-486a-92da-4d917aca7570" /> 
<img width="924" height="451" alt="image" src="https://github.com/user-attachments/assets/ce3a0290-bbc5-469a-8cf5-06874f62c909" />

</br>

<img width="920" height="388" alt="image" src="https://github.com/user-attachments/assets/87742188-4654-491a-9d75-0bc8e9221a05" /> 
<img width="924" height="451" alt="image" src="https://github.com/user-attachments/assets/759308d3-a7bf-4a8b-80a8-f13ee9f729d5" /> 


<img width="920" height="388" alt="image" src="https://github.com/user-attachments/assets/d9b93693-3a23-4e06-9145-1c0bdf1560d4" /> <img width="924" height="451" alt="image" src="https://github.com/user-attachments/assets/914e17bc-309d-4fb1-a9df-ad17bc686f56" />


**Video demostration**

![demo.gif](https://github.com/user-attachments/assets/0232c019-66df-4bf3-99f3-2ce261956642)

**Responsive design**

![responsive.gif](https://github.com/user-attachments/assets/85719991-4b72-4a82-8290-757a435353a5)


Fixes: #35555

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

whilstsomebody commented:
@apoorvapendse , are you up for the buddy review?

@zulipbot add "buddy review"

apoorvapendse commented on web/src/inbox_ui.ts:
```diff
@@ -1321,6 +1322,39 @@ export function collapse_or_expand(container_id: string): void {
+    update_collapsed_note_visibility();
+}
+
+export function count_collapsed_folders(): number {
+    return $("#inbox-list .inbox-folder.inbox-collapsed-state").length;
+}
+
+export function count_all_folders(): number {
+    return $("#inbox-list .inbox-folder:not(.hidden_by_filters)").length;
+}
```
Are these functions required, is there potential to reuse them?

Also, I am not sure why this is exported.

whilstsomebody replied:
I am not sure whether we would reuse them. But yeah, we can remove these for now.

I have removed them.

apoorvapendse reviewed (commented):
<img width="968" height="284" alt="Image" src="https://github.com/user-attachments/assets/b0f52df7-46bc-46fe-9213-f8990e3c18bd" />

Collapsed channels also count as collapsed sections right?
Should we show something for cases like these, where all the channels are collapsed but the folders are not.

To be precise, I don't understand the "Click on a *section* or *folder*" bit.

whilstsomebody commented:
> <img alt="Image" width="968" height="284" src="https://private-user-images.githubusercontent.com/102853901/473776244-b0f52df7-46bc-46fe-9213-f8990e3c18bd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQyMzM1NTcsIm5iZiI6MTc1NDIzMzI1NywicGF0aCI6Ii8xMDI4NTM5MDEvNDczNzc2MjQ0LWIwZjUyZGY3LTQ2YmMtNDZmZS05MjEzLWY4OTkwZTNjMThiZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgwM1QxNTAwNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05YmQ3N2IyOWY5MmE1Y2E5ZGRiYmVlYTA3ZjU4YTk3YmVkOWNkNWI2NWZlYzhkNmJjN2Q2YmVlOTU0NTNlMTA3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.kn99T0zYSfeRqIThgAnqj8lzUyrWHsMxhvWDnLQskuA">
> Collapsed channels also count as collapsed sections right? Should we show something for cases like these, where all the channels are collapsed but the folders are not.
> 
> To be precise, I don't understand the "Click on a _section_ or _folder_" bit.

Asked on the CZO.

whilstsomebody commented:
Current status of the PR:

- Updated the string as discussed in [#design > everything collapsed UI in inbox @ 💬](https://chat.zulip.org/#narrow/channel/101-design/topic/everything.20collapsed.20UI.20in.20inbox/near/2235843).
- Now the "Expand all" button expands folders as well as the channels.
- Addressed https://github.com/zulip/zulip/pull/35575#discussion_r2249933542

@apoorvapendse, could you take a look at the PR again? Thanks!

alya commented:
It also doesn't appear when I collapse all the folders:

<img width="1334" height="284" alt="Screenshot 2025-08-06 at 16 46 30@2x" src="https://github.com/user-attachments/assets/3ee1b2dd-ebda-4d41-90aa-68c87331afae" />

alya commented:
Please do more extensive self-review and manual testing here.

whilstsomebody commented:
I went through this case also and thought this is fine like this. But we can discuss this also.

apoorvapendse commented on web/src/inbox_ui.ts:
```diff
@@ -1323,6 +1324,50 @@ export function collapse_or_expand(container_id: string): void {
+    update_collapsed_note_visibility();
+}
+
+function update_collapsed_note_visibility(): void {
```
It would probably be helpful to add an explanatory comment about when we show or hide the notice.

whilstsomebody commented:
@apoorvapendse , i have addressed all the comments. Could you take a look?

sanchi-t reviewed (commented):
I am able to get into this state where I have no unreads but this message is showing below:

<img width="1919" height="960" alt="image" src="https://github.com/user-attachments/assets/d2512948-8cc1-415e-833e-d5e5b28d85b6" />

whilstsomebody commented:
@sanchi-t , actually the fallback case when there are no folders was not being checked. I have added that now.

<img width="1920" height="1200" alt="image" src="https://github.com/user-attachments/assets/b261cde0-ba18-46b8-afda-15bbcfcd8358" />

Please take a look.

alya commented:
@amanagr It would be great if you could review/finish up this PR by Tue morning in California, from whatever its state is.

timabbott commented on web/src/inbox_ui.ts:
```diff
@@ -1323,6 +1324,58 @@ export function collapse_or_expand(container_id: string): void {
+                .find(".inbox-header:not(.hidden_by_filters)");
+            return $channels.filter(":not(.inbox-collapsed-state)").length > 0;
+        }
+        return false;
```
I would expect this to use data structures like `collapsed_containers` rather than so much JQuery.

alya commented:
It would be great for the PR description to have both dark and light theme screenshots.

whilstsomebody commented:
@timabbott , i have now updated the code to use `collapsed_containers`.
@alya , I have added the light theme screenshots in the description.

timabbott commented:
With those changes, merged, thanks @whilstsomebody and @amanagr! I also made some changes to the string and commit message (which had a copy of the string in it).
